### DISCUSSION TITLE
M600

### DISCUSSION THREAD

**Discussion by gfrcr** [Guilouz/Creality-Helper-Script-Wiki#678] [Issues]
Good afternoon, I have a k1 max and I have been using the script since the beginning to have full control via klipper, however one thing that always bothered me was the m600, 95% of the time I print and I want to change the color with the m600. printer clogs, it's amazing, can anyone tell me what could be wrong or even what to do to prevent this from happening?

**Comment by ronoaldo** [comment]
I'm having the exact same issue, even with the pause command. I'm still investigating... I figured so far by looking at the printer settings and the OrcaSlicer settings:

1. The helper script has a modified version of the `gcode_macro QUIT_MATERIAL`, I'm testing right now the [factory one](https://github.com/CrealityOfficial/K1_Series_Klipper/blob/main/config/K1_MAX_CR4CU220812S12/gcode_macro.cfg#L67). The modified one seems to try to make a "tip" when retracting and this could be causing the clog.
2. OrcaSlicer has two features that I noticed that seems to interfere with the printer settings. One is to purge into the prime tower (it seems to create a large blob of filament after resuming) and another one for also creating a filament tip when changing filaments.

These are untested, I'm running another test print but I had to unclog the printer four times last time.

**Comment by ForgedDawning** [comment]
Hey I was having similar issues using the M600 cfg. The issue seems to be in the gcode_macro _UNLOAD_FILAMENT. The extruder movements are all over the place and just begging for a clog! I replaced the FOURTEEN separate Extruder commands with one extrude and one retract command as follows:
 G1 E50 F180
 G1 E-100 F180
 and it seems to be working well now! Your mileage may vary and you could even remove the first extrude before the retract that I have.

The trouble comes from trying to edit the Helper-script/M600-support.cfg as it's read only. The workaround I managed was to download the M600-support.cfg file, then change the Unload filament Macro values, then reupload the config file to the config root. Finally, you'll have to go to the printer.cfg and change [include Helper-Script/M600-support.cfg] to [include M600-support.cfg].

**Comment by mikiilis** [comment]
Hey! Same problem. I noticed that the difference between the standard macro `gcode_macro.cfg/QUIT_MATERIAL` from the firmware `1.3.3.46` and the macro `‎/Helper-Script/M600-support.cfg/_UNLOAD_FILAMENT` is in the line G1 **E10** F300. 
In `gcode_macro.cfg/QUIT_MATERIAL` it's  G1 **E100** F300 and with this value it works as expected for me. I also tried an in-between value E50, but I've had some difficulties getting the filament thread out, so I decided to stick with E100.
